### DISCUSSION TITLE
Does vector need read permissions for s3 sink ?

### DISCUSSION THREAD

**Discussion by mehrdad-khojastefar** [vectordotdev/vector#19792] [Q&A] [answered]
Hi, I have a vector instance that has an s3 sink, the s3 key associated to this vector instance only has write permissions.
Today I've found out that I am getting 403 from my s3 and by checking my s3 logs I've noticed that vector had tried to read something from s3.
The situation that this happened was not normal because alongside my s3 sink I have an elasticsearch sink and the elasticsearch was not responding at the moment of vector tried to read from s3. So I'm suspicious that maybe there is some fault-tolerance mechanism that needs read access.
I've reviewed the code for s3 sinks but have not found anything.

**Comment by jszwedko** [comment, marked as answer]
Hey!

The `aws_s3` uses `ListBucket` for the health check. We intended to publish the IAM requirements on the website, but, for now, you can see them described here: https://github.com/vectordotdev/vector/blob/0a2dc2bafa6e56218797a0c238118ed58fd94113/website/cue/reference/components/sinks/aws_s3.cue#L201-L218

**Comment by mehrdad-khojastefar** [comment]
Thanks, and how does it perform the health check ? What does it check ?

**Reply by jszwedko** [reply]
It executes a `ListBucket` API call.
